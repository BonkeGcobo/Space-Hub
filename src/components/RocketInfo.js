import React from 'react';
import PropTypes from 'prop-types';

function RocketInfo({ rocket_name: name, rocket_image: rocketImage, description }) {
  return (
    <div className="rocket">
      <img src={rocketImage} alt="rocket" width="250" />
      <div className="aboutRocket">
        <h2 className="rocket-name">{name}</h2>
        <p className="rocket-descrip">{description}</p>
        <button
          type="button"
          className="rocket-btn"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  );
}

RocketInfo.propTypes = {
  rocket_name: PropTypes.string.isRequired,
  rocket_image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default RocketInfo;
