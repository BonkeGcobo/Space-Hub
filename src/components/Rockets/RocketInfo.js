import React from 'react';
import PropTypes from 'prop-types';
import './Rocket.css';

function RocketInfo({
  id,
  rocket_name: name,
  rocket_image: rocketImage,
  description,
  reserved,
  handleToggleClick,
}) {
  return (
    <div className="rocket">
      <img src={rocketImage} alt="rocket" width="250" />
      <div className="aboutRocket">
        <h2 className="rocket-name">{name}</h2>
        <p className="reserve-sect">
          {reserved && (
            <span className="rev-word">
              Reserved
            </span>
          )}
          {' '}
          {description}
        </p>
        {reserved ? (
          <button
            type="button"
            className="cancel-book"
            onClick={() => handleToggleClick(id)}
          >
            Cancel Reservation
          </button>
        ) : (
          <button
            type="button"
            className="rocket-btn "
            onClick={() => handleToggleClick(id)}
          >
            Reserve Rocket
          </button>
        )}
      </div>
    </div>
  );
}

RocketInfo.propTypes = {
  id: PropTypes.number.isRequired,
  rocket_name: PropTypes.string.isRequired,
  rocket_image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  handleToggleClick: PropTypes.func.isRequired,
  reserved: PropTypes.bool,
};

RocketInfo.defaultProps = {
  reserved: false,
};

export default RocketInfo;
