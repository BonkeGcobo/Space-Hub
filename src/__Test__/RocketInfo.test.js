import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import RocketInfo from '../components/Rockets/RocketInfo';

const renderer = require('react-test-renderer');

it('should do render reserve rocket button and match the snapshot', () => {
  const id = 1;
  const rocketName = 'Test Rocket';
  const rocketImage = 'img';
  const description = 'Test';
  const reserved = false;
  const handleToggleClick = () => true;

  const { getByTestId } = render(<RocketInfo
    id={id}
    rocket_name={rocketName}
    rocket_image={rocketImage}
    description={description}
    reserved={reserved}
    handleToggleClick={handleToggleClick}
  />);
  expect(getByTestId('rocket-btn')).toHaveTextContent('Reserve Rocket');

  const tree = renderer.create(<RocketInfo
    id={id}
    rocket_name={rocketName}
    rocket_image={rocketImage}
    description={description}
    reserved={reserved}
    handleToggleClick={handleToggleClick}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
