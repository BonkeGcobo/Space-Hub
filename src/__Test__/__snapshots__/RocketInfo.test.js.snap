// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should do render reserve rocket button and match the snapshot 1`] = `
<div
  className="rocket"
>
  <img
    alt="rocket"
    src="img"
    width="250"
  />
  <div
    className="aboutRocket"
  >
    <h2
      className="rocket-name"
    >
      Test Rocket
    </h2>
    <p
      className="reserve-sect"
    >
       
      Test
    </p>
    <button
      className="rocket-btn"
      data-testid="rocket-btn"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
